### DISCUSSION TITLE
Finite Volume "Interior" Boundary Condition

### DISCUSSION THREAD

**Discussion by maxnezdyur** [idaholab/moose#19636] [Q&A Modules: Navier-Stokes] [answered]
Is there an "interior" boundary condition in Moose? I am trying to simulate flow through a 2D channel. In the middle of the channel, there is a side set. I want Moose to run for some time with the side set not as a boundary condition, and then at some point, I will turn it into a wall or some other boundary condition. I am getting the error that "The INSFVMomentumAdvection object mass is not completely bounded by INSFVBCs" if I don't include any boundary condition for that specific side set. The side set completely blocks the channel when it is turned on.

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can have an interior sideset but we currently dont allow interior boundary conditions in finite volume. It's not something that cant be done it's just that most of the time that will be an error in the input rather than something desirable.

There's multiple options for doing what you want to do. 
- restart the simulation at the point where you want to block the channel, loading up the previous velocity and pressure fields, with the mesh as you intend with a blockage
- have a volumetric blockage instead of a sideset. If you set a very high friction coefficient you should be able to essentially cut the flow. Note that if the flow has nowhere else to go it ll still go through, but with a very high pressure drop
- create an interface kernel that will penalize (with a lagrange multiplier) flow going through that sideset. Just a constraint on v_normal should do. 

Working with a boundary condition "that turns on" at some point seems difficult. The advection kernel examines at the beginning of the simulation all boundaries and determines which one it would execute on and with which behavior for Rhie Chow interpolation. This one would first be a normal region then become a wall, so the RC interpolation differs between those two cases.

@lindsayad might have a different opinion

Guillaume

**Reply by lindsayad** [reply]
Yea I think the easiest thing at this point would be to do the restart option that @GiudGiud recommended. After #18563 things would get a little simpler for this, but we still need to know what the flow boundaries are so that we can implicitly run advection kernels on outflow boundaries. Maybe in the future we can make that a Controllable parameter but that's not in the cards at the moment.
